File: services/server/src/safe_socket/safe_socket.py
import socket

def recv_all(socket: socket.socket, size):
    bytes = b""
    recived = 0
    while recived < size:
        try:
            byte_rec = socket.recv(size - recived)

            if byte_rec == 0:
                return b""

            recived += len(byte_rec)
            bytes += byte_rec
        except Exception as e:
            return b""

    return bytes

def send_all(socket: socket.socket, bytes):

    bytes_send = 0 
    while bytes_send < len(bytes):
        try:
            sended = socket.send(bytes[bytes_send:])
            bytes_send += sended
        except Exception as e:
            return 0

    return bytes_send

File: services/client/src/safe_socket/safe_socket.go
package safe_socket

import "io"

func SendAll(socket io.Writer, bytes []byte) error {

	bytes_send := 0
	for bytes_send < len(bytes){
		sended, err := socket.Write(bytes[bytes_send:])
		bytes_send += sended

		if err != nil {
			return err
		}
	}
	return nil
}

func RecvAll(socket io.Reader, size int) ([]byte, error) {
	bytes := make([]byte, size)

	recived := 0
	for recived < size {
		byte_rec, err := socket.Read(bytes[recived:])
		recived += byte_rec
		if err != nil {
			return nil, err
		}
	}

	return bytes, nil
}
